### PR TITLE
CircularBuffer: use vendored swift-collections/Deque instead

### PR DESCRIPTION
**Pull request by weissi**
Motivation:

Less (SwiftNIO maintained) code is better and Swift Collection's got an
excellent Deque implementation that NIO should just use.

Modification:

Use Swift Collection's Deque to back CircularBuffer instead rolling our
own (which is certainly much slower).

Result:

- Less (NIO maintained) code
- faster

### CODE REVIEW

**Comment by weissi** [comment]
CC @Lukasa  / @lorenty

File: Sources/NIO/VendoredDeque/Deque+CustomReflectable.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/* Changes for SwiftNIO
+   - renamed NIODeque to NIONIODeque (to prevent future clashes)
+   - made (NIO)NIODeque internal (not public)
```
Comment (Lukasa) [inline Sources/NIO/VendoredDeque/Deque+CustomReflectable.swift:3]: Something here is prefixing NIO again.

**Comment by ktoso** [comment]
Oh, awesome :-)

**Comment by weissi** [comment]
> Oh, awesome :-)

Except that it doesn’t work 😂😂. https://forums.swift.org/t/rfc-deque-should-it-have-an-opaque-index/47203

**Pull request closed without merging**